### ISSUE TITLE
Upgrading to v8 7.7 

### ISSUE THREAD

**Issue by SamSaffron** [rubyjs/libv8#281]
howdy @ignisf 

Looks like we are a few releases behind, any chance we can get this upgraded? Any work needed on mini racers side?

**Comment by ignisf** [comment]
will check

**Comment by ignisf** [comment]
There are a couple of compatibility issues again :(

```
mkdir -p tmp/x86_64-darwin18/mini_racer_extension/2.6.3
cd tmp/x86_64-darwin18/mini_racer_extension/2.6.3
/Users/petkobordjukov/.rbenv/versions/2.6.3/bin/ruby -I. ../../../../ext/mini_racer_extension/extconf.rb
checking for -lpthread... yes
checking for -lobjc... yes
creating Makefile
cd -
cd tmp/x86_64-darwin18/mini_racer_extension/2.6.3
/usr/bin/make
compiling ../../../../ext/mini_racer_extension/mini_racer_extension.cc
clang: warning: argument unused during compilation: '-rdynamic' [-Wunused-command-line-argument]
In file included from ../../../../ext/mini_racer_extension/mini_racer_extension.cc:2:
In file included from /Users/petkobordjukov/.rbenv/versions/2.6.3/include/ruby-2.6.0/ruby.h:33:
In file included from /Users/petkobordjukov/.rbenv/versions/2.6.3/include/ruby-2.6.0/ruby/ruby.h:2111:
/Users/petkobordjukov/.rbenv/versions/2.6.3/include/ruby-2.6.0/ruby/intern.h:56:19: warning: 'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
void rb_mem_clear(register VALUE*, register long);
                  ^~~~~~~~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/include/ruby-2.6.0/ruby/intern.h:56:36: warning: 'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
void rb_mem_clear(register VALUE*, register long);
                                   ^~~~~~~~~
../../../../ext/mini_racer_extension/mini_racer_extension.cc:235:53: error: no matching member function for call to 'Get'
            Local<Object> JSON = context->Global()->Get(String::NewFromUtf8(isolate, "JSON"))
                                 ~~~~~~~~~~~~~~~~~~~^~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3374:43: note: candidate function not viable: requires 2 arguments, but 1 was provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
                                          ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3377:43: note: candidate function not viable: requires 2 arguments, but 1 was provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
                                          ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:238:47: error: no matching member function for call to 'Get'
            Local<Function> stringify = JSON->Get(v8::String::NewFromUtf8(isolate, "stringify"))
                                        ~~~~~~^~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3374:43: note: candidate function not viable: requires 2 arguments, but 1 was provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
                                          ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3377:43: note: candidate function not viable: requires 2 arguments, but 1 was provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
                                          ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:239:25: error: 'Function' does not refer to a value
                    .As<Function>();
                        ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:4076:17: note: declared here
class V8_EXPORT Function : public Object {
                ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:239:35: error: expected expression
                    .As<Function>();
                                  ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:293:31: error: no viable conversion from 'MaybeLocal<v8::String>' to 'Local<v8::String>'
                Local<String> tmp = String::NewFromUtf8(isolate, "JavaScript was terminated (either by timeout or explicitly)");
                              ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:182:7: note: candidate constructor (the implicit copy constructor) not viable: no known
      conversion from 'MaybeLocal<v8::String>' to 'const v8::Local<v8::String> &' for 1st argument
class Local {
      ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:182:7: note: candidate constructor (the implicit move constructor) not viable: no known
      conversion from 'MaybeLocal<v8::String>' to 'v8::Local<v8::String> &&' for 1st argument
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:186:13: note: candidate template ignored: could not match 'Local' against 'MaybeLocal'
  V8_INLINE Local(Local<S> that)
            ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:398:39: error: no matching member function for call to 'Get'
          Local<Value> element = arr->Get(i);
                                 ~~~~~^~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3374:43: note: candidate function not viable: requires 2 arguments, but 1 was provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
                                          ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3377:43: note: candidate function not viable: requires 2 arguments, but 1 was provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
                                          ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:429:40: error: no matching member function for call to 'Get'
             Local<Value> key = props->Get(i);
                                ~~~~~~~^~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3374:43: note: candidate function not viable: requires 2 arguments, but 1 was provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
                                          ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3377:43: note: candidate function not viable: requires 2 arguments, but 1 was provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
                                          ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:431:48: error: no matching member function for call to 'Get'
             Local<Value> prop_value = object->Get(key);
                                       ~~~~~~~~^~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3374:43: note: candidate function not viable: requires 2 arguments, but 1 was provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
                                          ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3377:43: note: candidate function not viable: requires 2 arguments, but 1 was provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
                                          ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:503:14: error: no matching member function for call to 'Set'
      array->Set(i, convert_ruby_to_v8(isolate, context, rb_ary_entry(value, i)));
      ~~~~~~~^~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3327:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                    ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3330:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
                                    ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:512:14: error: no matching member function for call to 'Set'
            object->Set(convert_ruby_to_v8(isolate, context, rb_ary_entry(pair, 0)),
            ~~~~~~~~^~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3327:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                    ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3330:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
                                    ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:542:20: error: no matching member function for call to 'Escape'
      return scope.Escape(String::NewFromUtf8(isolate, "Undefined Conversion"));
             ~~~~~~^~~~~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:1036:22: note: candidate function not viable: no known conversion from
      'MaybeLocal<v8::String>' to 'internal::Address *' (aka 'unsigned long *') for 1st argument
  internal::Address* Escape(internal::Address* escape_value);
                     ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:1014:22: note: candidate template ignored: could not match 'Local' against 'MaybeLocal'
  V8_INLINE Local<T> Escape(Local<T> value) {
                     ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:1018:44: error: no viable conversion from 'MaybeLocal<v8::String>' to 'Local<v8::Value>'
        args->GetIsolate()->ThrowException(String::NewFromUtf8(args->GetIsolate(), "Terminated execution during transition from Ruby to JS"));
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:182:7: note: candidate constructor (the implicit copy constructor) not viable: no known
      conversion from 'MaybeLocal<v8::String>' to 'const v8::Local<v8::Value> &' for 1st argument
class Local {
      ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:182:7: note: candidate constructor (the implicit move constructor) not viable: no known
      conversion from 'MaybeLocal<v8::String>' to 'v8::Local<v8::Value> &&' for 1st argument
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:186:13: note: candidate template ignored: could not match 'Local' against 'MaybeLocal'
  V8_INLINE Local(Local<S> that)
            ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:7968:44: note: passing argument to parameter 'exception' here
  Local<Value> ThrowException(Local<Value> exception);
                                           ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:1032:44: error: no viable conversion from 'MaybeLocal<v8::String>' to 'Local<v8::Value>'
        args->GetIsolate()->ThrowException(String::NewFromUtf8(args->GetIsolate(), "Ruby exception"));
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:182:7: note: candidate constructor (the implicit copy constructor) not viable: no known
      conversion from 'MaybeLocal<v8::String>' to 'const v8::Local<v8::Value> &' for 1st argument
class Local {
      ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:182:7: note: candidate constructor (the implicit move constructor) not viable: no known
      conversion from 'MaybeLocal<v8::String>' to 'v8::Local<v8::Value> &&' for 1st argument
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:186:13: note: candidate template ignored: could not match 'Local' against 'MaybeLocal'
  V8_INLINE Local(Local<S> that)
            ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:7968:44: note: passing argument to parameter 'exception' here
  Local<Value> ThrowException(Local<Value> exception);
                                           ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:1102:32: error: no matching member function for call to 'Set'
            context->Global()->Set(
            ~~~~~~~~~~~~~~~~~~~^~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3327:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                    ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3330:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
                                    ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:1125:45: error: no matching member function for call to 'Set'
                        value.As<Object>()->Set(
                        ~~~~~~~~~~~~~~~~~~~~^~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3327:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                    ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:3330:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
                                    ^
../../../../ext/mini_racer_extension/mini_racer_extension.cc:1506:23: error: no viable conversion from 'MaybeLocal<v8::String>' to 'Local<v8::String>'
        Local<String> fname = String::NewFromUtf8(isolate, call.function_name);
                      ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:182:7: note: candidate constructor (the implicit copy constructor) not viable: no known
      conversion from 'MaybeLocal<v8::String>' to 'const v8::Local<v8::String> &' for 1st argument
class Local {
      ^
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:182:7: note: candidate constructor (the implicit move constructor) not viable: no known
      conversion from 'MaybeLocal<v8::String>' to 'v8::Local<v8::String> &&' for 1st argument
/Users/petkobordjukov/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/libv8-7.7.299.11.0beta1-x86_64-darwin-18/vendor/v8/include/v8.h:186:13: note: candidate template ignored: could not match 'Local' against 'MaybeLocal'
  V8_INLINE Local(Local<S> that)
            ^
2 warnings and 16 errors generated.
make: *** [mini_racer_extension.o] Error 1
rake aborted!
Command failed with status (2): [/usr/bin/make...]
/Users/petkobordjukov/.rbenv/versions/2.6.3/bin/bundle:23:in `load'
/Users/petkobordjukov/.rbenv/versions/2.6.3/bin/bundle:23:in `<main>'
Tasks: TOP => default => compile => compile:x86_64-darwin18 => compile:mini_racer_extension:x86_64-darwin18 => copy:mini_racer_extension:x86_64-darwin18:2.6.3 => tmp/x86_64-darwin18/mini_racer_extension/2.6.3/mini_racer_extension.bundle
(See full trace by running task with --trace)
[petkobordjukov@SOF-PETBOR-MAC mini_racer]$
```

**Comment by SamSaffron** [comment]
Oh no!!! scary!

**Comment by nightpool** [comment]
Closed via #301 (pending rubygems release)
